enhancements/quota-management.md

```diff
@@ -0,0 +1,103 @@
+aims to contain and provide a smart solution to tackle storage related problems for Quay customers.   
+In the first phase i.e, Quota Reporting: a super admin can see the storage consumption of all of their organizations, and a user can see the storage consumption of their organization.
+In the second phase i.e, Quota Management: a superuser can define soft and hard checks for Quay users.
+A soft check implies that the Quay user will be notified if the storage consumption of an organization reaches the configured threshold.
+A hard check implies that the Quay user will not be allowed to push to the registry once the storage consumption reaches the set limit.
+In the third phase, a quay admin can configure auto pruning of the images based on when they were last used to free up quota.
+
+## Motivation
+
+Users can create new organizations and repositories in their home organizations without any practical storage limitation.
+On-prem Quay customers are subject to capacity limits of their environment, specifically backend storage consumption.
+No knowledge about their storage consumption makes it hard for a service owner of a Quay registry to define SLAs and maintain a budget.
+
+### Goals
+
+* A Quay superuser can see the overall storage consumption of the entire registry i.e., an overview of all organizations depicting the aggregate storage consumption of images in an organization.
+* A Quay user can define soft quota limit on an organization that will trigger a warning when the set limit has reached.
+* A Quay user can configure alerts that will allow Quay to send notifications when set storage threshold has reached.
+* A Quay user can define hard quota limit on an organization which will not allow them to push any more images when set threshold has reached.
+
+### Non-Goals
+
+* We will not be implementing shared layer calculation i.e., if a user pushes two different versions of an image and there are some common layers between the two,
+the storage consumption is calculated independently for each of the common layers.
+
+## Design Details
+
+Initial work is to add Quota Reporting on organizations i.e., providing Quay users with their storage consumption data on a Quay organization.
+In the next steps, we will add Quota Enforcements i.e., soft and hard checks that notify or restrict Quay users from pushing images to registry.
+
+The expected design flow is depicted as below:
+![](https://user-images.githubusercontent.com/11522230/144654849-62d00b53-3be7-4009-afb6-2523f2931841.jpeg)
+NOTE: The boxes with the black border show the current flow, and the boxes with the green border show what needs to be implemented to support this feature. Design credits: @kwestpha
+
+The table RepositorySize (to be created) stores the storage consumption(in KBs) of a Quay repository within an organization. The sum of all repository sizes for an organization,
```

Comment:
We should keep track of sizes in bytes, just for consistency, since that's the unit we use elsewhere in Quay. Any conversion if needed is easy enough to do.  A bigint should be more than enough to cover any reasonable use cases.